angel_system/data/common/kwcoco_utils.py

```diff
@@ -105,7 +106,17 @@ def add_activity_gt_to_kwcoco(topic, task, dset, activity_config_fn):
                 label = "background"
                 activity_label = label
             else:
-                label = matching_gt.iloc[0]["class_label"]
+                label = matching_gt.iloc[0]["class_label"] # default to the first gt
+
+                # Hacky temporary fix
+                # In the medical data, step 1 can cover the same frames as other steps,
+                # So we attempt to find the "hidden" step instead
+                if len(matching_gt) > 1:
+                    possible_labels = []
+                    for i, row in matching_gt.iterrows():
+                        possible_labels.append(row["class_label"])
+                    label = max(possible_labels)
```

Comment:
Checking that this is still relevant and the comment is up to date?

Reply:
Updated wording a bit - I think still relevant, this is fixing the issue that we may get step 1 AND step x as gt for a singular frame. I still think it's hacky though because it assumes an integer label. This will break with non integer labels and if the problematic class is a higher step id.